Type 'null' is not assignable to type 'ResourceIdentity | ResourceIdentity[] | undefined'.

Trying to make a relationship empty in the JSONAPI serializer:

```js
  deserializeRelationship(record: Record, relationship: string, value: ResourceRelationship): void {
    if (value && value.data && !Array.isArray(value.data) && value.data.id === '-1') {
      value.data = null;
    }
```

Seeing this about `.data`:

> error TS2322: Type 'null' is not assignable to type 'ResourceIdentity | ResourceIdentity[] | undefined'.

According to the JSONAPI spec, relationship can be empty, indicated by `data: null`.

CC @simonihmig

--- a/packages/@orbit/jsonapi/src/jsonapi-document.ts
+++ b/packages/@orbit/jsonapi/src/jsonapi-document.ts
@@ -6,7 +6,7 @@ export interface ResourceIdentity {
 }
 
 export interface ResourceHasOneRelationship {
-  data?: ResourceIdentity;
+  data?: ResourceIdentity | null;
   meta?: any;
 }
 

